packages/@react-spectrum/dropzone/docs/DropZone.mdx

```diff
@@ -0,0 +1,338 @@
+
+```tsx example
+function Example() {
+  let [isFilled, setIsFilled] = React.useState(false);
+
+  return (
+    <>
+      <Draggable />
+      <DropZone
+        maxWidth="size-3000"
+        isFilled={isFilled}
+        onDrop={() => setIsFilled(true)}>
+        <IllustratedMessage>
+          <Upload />
+          <Heading>
+            <Text slot="label">
+              Drag and drop here
+            </Text>
+          </Heading>
+        </IllustratedMessage>
+      </DropZone>
+    </>
+  );
+}
+```
+
+The `Draggable` component used above is defined below. See [useDrag](../react-aria/useDrag.html) for more details and documentation.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example render=false export=true
+import {useDrag} from '@react-aria/dnd';
+
+function Draggable() {
+  let {dragProps, isDragging} = useDrag({
+    getItems() {
+      return [{
+        'text/plain': 'hello world',
+        'my-app-custom-type': JSON.stringify({message: 'hello world'})
+      }];
+    }
+  });
+
+  return (
+    <div {...dragProps} role="button" tabIndex={0} className={`draggable ${isDragging ? 'dragging' : ''}`}>
+      Drag me
+    </div>
+  );
+}
+```
+</details>
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.draggable {
+  display: inline-block;
+  vertical-align: top;
+  border: 1px solid gray;
+  padding: 10px;
+  margin-right: 20px;
+  margin-bottom: 20px;
+  border-radius: 4px;
+  height: fit-content;
+}
+
+.draggable.dragging {
+  opacity: 0.5;
+}
+```
+
+</details>
+
+
+## Props
+
+<PropTable component={docs.exports.DropZone} links={docs.links} />
+
+## Visual options
+
+### Filled state
+
+The user is responsible for both managing the filled state of a drop zone and handling the associated styling. To set the drop zone in a filled state, the user must pass the `isFilled` prop.
+
+The example below demonstrates one way of styling the filled state.
+
+```tsx example
+function Example() {
+  let [filledSrc, setFilledSrc] = React.useState(null);
+  let [isFilled, setIsFilled] = React.useState(false);
+
+  return (
+    <>
+      <DraggableImage />
+      <DropZone
+        isFilled={isFilled}
+        maxWidth="size-3000"
+        height="size-2400"
+        UNSAFE_className={isFilled && 'is-filled'}
+        getDropOperation={(types) =>  (types.has('image/png') || types.has('image/jpeg')) ? 'copy' : 'cancel'}
+        onDrop={async (e) => {
+          e.items.find(async (item) => {
+            if (item.kind === 'file') {
+              if (item.type === 'image/jpeg' || item.type === 'image/png') {
+                setFilledSrc(URL.createObjectURL(await item.getFile()));
+                setIsFilled(true);
+              }
+            } else if (item.kind === 'text') {
+              setFilledSrc(await item.getText('image/jpeg'));
+              setIsFilled(true);
+            }
```

Comment:
just curious, is this portion of the code the one being hit when using `DraggableImage`?

Reply:
yup that's correct...not sure if there's a better way to handle it but open to changing it

Reply:
I think it is fine as is. If we are worried about this code block being a bit too specific to how DraggableImage is implemented, we could just rely on the user dropping their own png/jpegs and just prompt them to do so via a blurb of "Try dropping a png/jpeg into the dropzone below" in the sentence before the example IMO